cupy/_core/_dtype.pyx

```diff
@@ -124,3 +124,41 @@ cpdef void _raise_if_invalid_cast(
     raise TypeError(
         f'Cannot cast {argname} from {from_dt!r} to {to_dt!r} '
         f'according to the rule \'{casting}\'')
+
+
+@cython.profile(False)
+cdef void populate_format(Py_buffer* buf, str dtype) except*:
+    if dtype == "?":
+        buf.format = '?'
+    elif dtype == "b":
+        buf.format = 'b'
+    elif dtype == "h":
+        buf.format = 'h'
+    elif dtype == "i":
+        buf.format = 'i'
+    elif dtype == "l":
+        buf.format = 'l'
+    elif dtype == "q":
+        buf.format = 'q'
+    elif dtype == "B":
+        buf.format = 'B'
+    elif dtype == "H":
+        buf.format = 'H'
+    elif dtype == "I":
+        buf.format = 'I'
+    elif dtype == "L":
+        buf.format = 'L'
+    elif dtype == "Q":
+        buf.format = 'Q'
+    elif dtype == "e":
+        buf.format = 'e'
+    elif dtype == "f":
+        buf.format = 'f'
+    elif dtype == "d":
+        buf.format = 'd'
+    elif dtype == "F":
+        buf.format = 'Zf'
+    elif dtype == "D":
+        buf.format = 'Zd'
```

Comment:
It might be worth putting this in a `dict` and just using that here

For example, in this Cython code, we added [a `dict` for lookups]( https://github.com/rapidsai/ucx-py/blob/5ceff39ee6069f03a80e054a2c612c8d9c45edc3/ucp/_libs/arr.pyx#L32 ), which we use [in this block]( https://github.com/rapidsai/ucx-py/blob/5ceff39ee6069f03a80e054a2c612c8d9c45edc3/ucp/_libs/arr.pyx#L104-L113 )

This is actually pretty performant especially when Python `intern`s the strings because then the hash of the `str` is known and can be used to grab the key right away

Reply:
Done.